src/module.c

```diff
@@ -517,13 +517,16 @@ int ReplySeriesRange(RedisModuleCtx *ctx, Series *series, api_timestamp_t start_
     			max(start_ts, series->lastTimestamp - series->retentionTime) : start_ts;
     }
     SeriesIterator iterator = SeriesQuery(series, start_ts, end_ts);
+    if (iterator.chunkIterator == NULL) { 
```

Comment:
This is not a hard stop, but it looks like a leaky abstraction, technically at this level you shouldn't even see the chunk iterator and it should be opaque.
Another way to handle this would be just  responding with `CR_END` on the first `SeriesIteratorGetNext`, this way you don't have a special case for that and you don't touch the chunks at this level.

Reply:
I was considering a return value instead.
I will make the change and send you.